core/quivr_core/rag/entities/config.py

```diff
@@ -0,0 +1,452 @@
+                    return supplier
+        # Return None if no supplier matches the model name
+        return None
+
+    @classmethod
+    def get_llm_model_config(
+        cls, supplier: DefaultModelSuppliers, model_name: str
+    ) -> Optional[LLMConfig]:
+        """Retrieve the LLMConfig (context and tokenizer_hub) for a given supplier and model."""
+        supplier_defaults = cls._model_defaults.get(supplier)
+        if not supplier_defaults:
+            return None
+
+        # Use startswith logic for matching model names
+        for key, config in supplier_defaults.items():
+            if model_name.startswith(key):
+                return config
+
+        return None
+
+
+class LLMEndpointConfig(QuivrBaseConfig):
+    supplier: DefaultModelSuppliers = DefaultModelSuppliers.OPENAI
+    model: str = "gpt-4o"
+    context_length: int | None = None
+    tokenizer_hub: str | None = None
+    llm_base_url: str | None = None
+    env_variable_name: str | None = None
+    llm_api_key: str | None = None
+    max_context_tokens: int = 2000
+    max_output_tokens: int = 2000
+    temperature: float = 0.7
+    streaming: bool = True
+    prompt: CustomPromptsModel | None = None
+
+    _FALLBACK_TOKENIZER = "cl100k_base"
+
+    @property
+    def fallback_tokenizer(self) -> str:
+        return self._FALLBACK_TOKENIZER
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.set_llm_model_config()
+        self.set_api_key()
+
+    def set_api_key(self, force_reset: bool = False):
+        if not self.supplier:
+            return
+
+        # Check if the corresponding API key environment variable is set
+        if not self.env_variable_name:
+            self.env_variable_name = (
+                f"{normalize_to_env_variable_name(self.supplier)}_API_KEY"
+            )
+
+        if not self.llm_api_key or force_reset:
+            self.llm_api_key = os.getenv(self.env_variable_name)
+
+        if not self.llm_api_key:
+            logger.warning(f"The API key for supplier '{self.supplier}' is not set. ")
+
+    def set_llm_model_config(self):
+        # Automatically set context_length and tokenizer_hub based on the supplier and model
+        llm_model_config = LLMModelConfig.get_llm_model_config(
+            self.supplier, self.model
+        )
+        if llm_model_config:
+            self.context_length = llm_model_config.context
+            self.tokenizer_hub = llm_model_config.tokenizer_hub
+
+    def set_llm_model(self, model: str):
+        supplier = LLMModelConfig.get_supplier_by_model_name(model)
+        if supplier is None:
+            raise ValueError(
+                f"Cannot find the corresponding supplier for model {model}"
+            )
+        self.supplier = supplier
+        self.model = model
+
+        self.set_llm_model_config()
+        self.set_api_key(force_reset=True)
+
+    def set_from_sqlmodel(self, sqlmodel: SQLModel, mapping: Dict[str, str]):
```

Comment:
We don't need the SQLModel here we can take a BaseModel. We can then remove the sqlmodel dep

Reply:
done